src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,295 @@
+        System.out.println("\tBut anyways bye! Hope to see you again soon, friend! (ɔ◔‿◔)ɔ ♥");
+        System.out.println();
+        printDivider();
+    }
+
+    public static void printExceptionMessage(String exceptionType) {
+        printDivider();
+
+        switch (exceptionType) {
+        case EXCEPTION_1 :
+            //Empty description of a task
+            System.out.println("\t☹ OOPS!!! The description of a task cannot be empty...");
+            break;
+        case EXCEPTION_2 :
+            //Commands other than the ones recognized
+            System.out.println("\t☹ OOPS!!! I'm sorry, but I don't know what that means...");
+            break;
+        case EXCEPTION_3 :
+            //Mark/Unmark/Delete index out of bounds
+            System.out.println("\t☹ OOPS!!! Please provide a valid task index...");
+        case EXCEPTION_4:
+            //File not found
+            System.out.println("\t☹ OOPS!!! Unfortunately file is not found...");
+        }
+
+        System.out.println();
+        printDivider();
+    }
+
+    public static void readFile() throws FileNotFoundException {
+        File file = new File("duke.txt");
+        try {
+            boolean isExisting = file.createNewFile();
+        } catch (IOException e) {
+            printExceptionMessage(EXCEPTION_4);
+        }
+        Scanner scanner = new Scanner(file);
+
+        while (scanner.hasNext()) {
+            String line = scanner.nextLine();
+            String[] words = line.split(" ");
+            try {
+                Task currentTask = createTask(line.substring(2));
+                addTask(currentTask, true);
+                if (words[0].equals("1")) {
+                    markAsDone(String.valueOf(numberOfTasks), true);
+                }
+            } catch (DukeException e) {
+                printExceptionMessage(EXCEPTION_2);
+            }
+        }
+    }
+
+    public static void appendToFile(String command) throws IOException {
+        FileWriter fw = new FileWriter("duke.txt", true);
+        fw.write("0 " + command + "\n");
+        fw.close();
+    }
+
+    public static void rewriteFile() throws IOException {
+        FileWriter fw = new FileWriter("duke.txt");
+        String number;
+        for (Task task : tasks) {
+            if (task.isDone) {
+                number = "1";
+            } else {
+                number = "0";
+            }
+            if (task instanceof Todo) {
+                fw.write(number + " todo " + task.description + "\n");
+            } else if (task instanceof Deadline) {
+                fw.write(number + " deadline " + task.description + " /by " + ((Deadline) task).by + "\n");
+            } else if (task instanceof Event) {
+                fw.write(number + " event " + task.description + " /at " + ((Event) task).at + "\n");
+            }
+        }
+        fw.close();
+    }
+
+    public static void performAction(String command) throws DukeException{
```

Comment:
The naming of this method could be clearer, as it might be confusing what performAction means - what action is actually being performed. For example, executeUserCommand() would be a better alternative.

Code Quality Guideline: Name well and use name to explain